### PR TITLE
Fix the usage of `CallBuilder` pattern.

### PR DESCRIPTION
**Pull request by deuszx**
Respective entry in the migration guide has been added - https://www.notion.so/cardinalcryptography/Contracts-e9fcab6bc8094707ade4d243ea69994a#fb776d2f54ba4099acb8a04c185d7bca

### CODE REVIEW

**Review by obrok — APPROVED** [review]

File: contracts/bulletin_board/lib.rs [hunk 1]
```diff
@@ -444,11 +454,23 @@ mod bulletin_board {
...
+                    Ok(Err(lang_error)) => {
+                        panic!("Unexpected ink::LangError: {:?}", lang_error)
+                    }
+                    Ok(Ok(Err(contract_call_error))) => {
+                        return Err(BulletinBoardError::from(
+                            contract_call_error,
+                        ))
+                    }
+                    Ok(Ok(Ok(_unit))) => return Ok(()),
+                }
```
[the remark this answers is not in the document]
Comment (deuszx) [reply contracts/bulletin_board/lib.rs:473]: Yes, the match here is supposed to be an explanation of what kind of errors are being returned.

File: contracts/bulletin_board/lib.rs [hunk 2]
```diff
@@ -432,7 +438,11 @@ mod bulletin_board {
             cost: u128,
         ) -> Result<(), BulletinBoardError> {
             if let Some(highlight_board) = self.highlighted_posts_board {
-                build_call::<DefaultEnvironment>()
+                // Since the introduction of `ink::LangError` we got a new
+                // "channel" where the contract errors may be pushed into.
+                // In this particular example we have three layers of `Result`
+                // types.
```
[the remark this answers is not in the document]
Comment (deuszx) [reply contracts/bulletin_board/lib.rs:444]: I will simplify the comment and remove ambiguity.

File: contracts/bulletin_board/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (deuszx) [reply contracts/bulletin_board/lib.rs:473]: This is a contract though, so I try to return some kind of error that can be (de)serialized. Only in the case of `lang_error` I'm panicing as there's no way to handle that IIUC. [Here](https://github.com/paritytech/ink/pull/1525) there's some work where `LangError` will, by default, not be handled and cause panic as well.

File: contracts/bulletin_board/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (deuszx) [reply contracts/bulletin_board/lib.rs]: Rephrased in https://github.com/Cardinal-Cryptography/bulletin-board-example/pull/10/commits/a44181bbbe84c16c807329c79e3b9c613d6cf4f1

**Pull request merged**